### ISSUE TITLE
[Accessibility]: Clicking on "Mix precise and search-based results" doesn't announce anything 

### ISSUE THREAD

**Issue by varungandhi-src** [sourcegraph/sourcegraph#36570]
### Audit type

Screen reader navigation

### User journey audit issue

https://github.com/sourcegraph/sourcegraph/issues/33510

### Problem description

Performing a screen reader click on the "Mix precise and search-based results" link doesn't announce any change in state.

Similarly for the "Hide search-based results when precise results are available" link which replaces it after the click.

### Expected behavior

There should be some announcement describing what happened at a high-level.

### Assigning labels

- [ ] Please give this issue an estimate by applying a label like `estimate/Xd`, where X is the estimated number of days it will take to complete.
- [X] If this issue is specific to a specific Sourcegraph product, please assign the appropriate team label to this issue.
- [ ] If this issue will require input from designers in order to complete, please assign the label `needs-design`.
- [ ] If you are confident that this issue should be fixed by GitStart, please assign the label `gitstart`.

### Owner

This issue will be fixed by my team, I have assigned a relevant member to it, or I will do so in the near future.
